ffmulticonverter/ffmulticonverter.py

```diff
@@ -238,13 +238,9 @@ def check_for_dependencies(self):
         self.unoconv = utils.is_installed('unoconv')
         self.imagemagick = utils.is_installed('convert')
 
-        missing = []
-        if not self.ffmpeg_path:
-            missing.append('ffmpeg')
-        if not self.unoconv:
-            missing.append('unoconv')
-        if not self.imagemagick:
-            missing.append('imagemagick')
+        missing = [dependency for dependency in
+                   ('ffmpeg', 'unoconv', 'convert') if not
+                   utils.is_installed(dependency)]
```

Comment:
Hmm. The problem with this is that utils.is_installed has already been called for all of the dependencies, so it doesn't make sense to call it again. We also need the self.unoconv etc. flags for use in other parts of the program. If there were more than 3 dependencies I agree, it would probably worth it to try structure it in another way, but in this case I think it's fine.

Reply:
What about this:
# I suppose this checks if self.ffmpeg_path was in the settings file, so I think is redundant to test using utils.is_installed(self.ffmpeg_path)
if not self.ffmpeg_path: # This must be True if self.ffmpeg_path == ''
            self.ffmpeg_path = utils.is_installed('ffmpeg')  # So I assign the value return by utils.is_installed()
            QSettings().setValue('ffmpeg_path', self.ffmpeg_path)
        self.unoconv = utils.is_installed('unoconv')
        self.imagemagick = utils.is_installed('convert')

        missing = [dependency for dependency in
                   (self.ffmpeg_path, self.unoconv, self.imagemagick) if not
                   dependency]

Reply:
We need to check if this path for ffmpeg indeed exist, that's why we call is_installed().

Also, self.unoconv, self.imagemagick etc. have boolean True/False values, so in the way you wrote it, the missing variable will be a list of Booleans. But we want it to contain the names (strings) of the missing dependencies instead.

Reply:
Well, there must be something wrong... utils.is_installed return fpath if the program is installed or '' otherwise, not True or False. So I don't get the logic

Reply:
Well you really get an empty list if dependencies are not installed. If some dependency is not install and other are, you will get:
missing = [self.ffmpeg,...]
and that is not what you want... ok

Reply:
Nope. Remember, utils.is_installed() returns an empty string for every dependency that is missing.

So what you get with the above list comprehension is:
```
>>> ffmpeg_path = ''
>>> unoconv = ''
>>> imagemagick = ''
>>> [dep for dep in (ffmpeg_path, unoconv, imagemagick) if not dep]
['', '', '']
>>> unoconv = '/path/to/unoconv'
>>> [dep for dep in (ffmpeg_path, unoconv, imagemagick) if not dep]
['', '']
>>> 
```
Anyway, in either case, we just need the string 'ffmpeg', 'unoconv', etc. to be appended in the variable missing for every dependency that is missing.

Reply:
Oh, you are right now... ok, I think I will be around for some more suggestions. I just need to study more deeply the code.
I have a general suggestion. Why don't you use PEP8?
I think that Pylint could be nice on this...

Reply:
In general I follow PEP8, just not every tiny detail. 
It's true I didn't run code analysis tools on this (I think pylint is nice, but PEP8 is way too strict and kind of annoying sometimes). But in any way I don't think that the current state is that bad that need drastic changes.

I would prefer to change the structure of the code and add unit tests instead, but unfortunately I'll never have the time for this.

Reply:
I would like to help the project cause I have been using it myself. In general I think PEP8 has some annoying things but it is a good guide.
I would like to help on changing the structure of the code and maybe use the MVC pattern. I think in the actual state it is very difficult to create unittests, cause things are very mixed.
I hope you to forget my English, I am not native.

Reply:
That's great, any contributions are welcome! I also wanted to use something like MVC, but as I said I never found the time for it. I'd be glad to help others though and do reviews. I don't think it's difficult to write unittests, just it has to be made gradually (maybe at the same time as restructuring the code). At least its a small project without lots of code. To make it even smaller and more manageable (and more MVC-like) I was thinking of using QtDesigner for producing xml code for windows and dialogs instead of directly have python code (like we do now).